src/test/java/org/openmrs/contrib/qaframework/automation/ClinicalVisitSteps.java

```diff
@@ -170,7 +207,18 @@ public void saveCondition() {
 	@Then("the system adds New Condition in Conditions table")
 	public void systemAddsCondition() {
 		assertNotNull(conditionsPage.getConditionsList());
+	}
+
+	@When("the user clicks on the delete button from dashboard")
+	public void deleteCondition() {
+		conditionsPage.deleteFirstActive();
+	}
+
+	@Then("user clicks on the yes button to confirm")
+	public void confirmDeleteCondition() {
+		conditionsPage.confirmDeleteCondition();
 		dashboardPage = conditionsPage.clickReturn();
+
 	}
```

Comment:
deleteCondition() would be better to accomplish the intended action. That would simply mean you do not need to create another method confirmDeleteConditon() but add its implementation in the deleteCondition().
```js
       @And("a user clicks on the delete button from dashboard")
	public void deleteCondition() {
	    conditionsPage.deleteFirstActive();
            conditionsPage.confirmDeleteCondition();
	    dashboardPage = conditionsPage.clickReturn();
	}